modules/globebrowsing/tile/rawtiledatareader/rawtiledatareader.cpp

```diff
@@ -0,0 +1,416 @@
+            // +-+-----+               +-+-----+        |
+            //                           |              | <-- io.read.fullRegion  
+            //                           |              |
+            //                           +--------------+
+
+            if (cutoff.read.region.area() > 0) {
+                // Wrap by repeating
+                PixelRegion::Side oppositeSide = (PixelRegion::Side) ((i + 2) % 4);
+
+                cutoff.read.region.align(
+                    oppositeSide, io.read.fullRegion.edge(oppositeSide));
+
+                // Example:
+                // The cut off region is wrapped to the opposite side of the region,
+                // i.e. "repeated". Note that we don't want WRITE region to change, 
+                // we're only wrapping the READ region.
+
+                // cutoff.write.region   io.read.region cutoff.read.region
+                //  |  io.write.region        |          |
+                //  |   |                     V          V
+                //  V   V                  +-----+      +-+
+                // +-+-----+               |     |------| |
+                // | |     |               |     |      | | 
+                // | |     |               |     |      | |
+                // | |     |               +-----+      +-+
+                // +-+-----+               |              | <-- io.read.fullRegion  
+                //                         |              |
+                //                         +--------------+
+
+                // Example:
+                // The cutoff region has been repeated along one of its sides, but 
+                // as we can see in this example, it still has a top part outside the
+                // defined gdal region. This is handled through recursion.
+                RawTile::ReadError err = repeatedRasterRead(
+                    rasterBand, cutoff, dataDestination, depth + 1);
+
+                worstError = std::max(worstError, err);
+            }
+        }
+    }
+        
+    RawTile::ReadError err = rasterRead(rasterBand, io, dataDestination);
+
+    // The return error from a repeated rasterRead is ONLY based on the main region,
+    // which in the usual case will cover the main area of the patch anyway
+    return err;
+}
+
+std::shared_ptr<TileMetaData> RawTileDataReader::getTileMetaData(
+    std::shared_ptr<RawTile> rawTile, const PixelRegion& region)
+{
+    ensureInitialized();
+    size_t bytesPerLine = _dataLayout.bytesPerPixel * region.numPixels.x;
+
+    TileMetaData* preprocessData = new TileMetaData();
+    preprocessData->maxValues.resize(_dataLayout.numRasters);
+    preprocessData->minValues.resize(_dataLayout.numRasters);
+    preprocessData->hasMissingData.resize(_dataLayout.numRasters);
+        
+    std::vector<float> noDataValues;
+    noDataValues.resize(_dataLayout.numRasters);
+
+    for (size_t c = 0; c < _dataLayout.numRasters; c++) {
```

Comment:
For all in this function:
In general, it is better to use `++i` then `i++` as the first one might translate into only a single machine instruction